doc/howto/homebrew-package.rst

```diff
@@ -0,0 +1,83 @@
+How to make a Homebrew Package with Dune
+========================================
+
+This guide will show you how to make a Homebrew package for an application
+using Dune package management. The only dependency of the Homebrew package will
+be Dune, and all OCaml dependencies (including the OCaml compiler) will be
+installed by Dune while building the Homebrew package.
+
+To use Dune package management to build a project as a Homebrew package, the
+project must have a source archive hosted online somewhere (e.g. a gzipped
+tarball on the project's github release page), and this archive must contain a
+lockdir completely specifying the project's dependencies. Such a lockdir can be
+generated by running ``dune pkg lock``.
+
+Let's make a brew package!
+
+First, familiarize yourself with Homebrew's terminology and packaging conventions
+`here <https://docs.brew.sh/Adding-Software-to-Homebrew>`_.
+
+Homebrew packages are recommended to be source-based, and for the source code
+to be explicitly versioned, so for this example assume ``my_app`` has a
+versioned archive hosted on Github with version ``0.1.0``.
+
+Homebrew can generate a starting point for a formula if you point it at a
+source archive hosted on Github:
+
+.. code:: console
+
+  $ brew create https://github.com/me/my_app/archive/refs/tags/0.1.0.tar.gz
+
+A source archive like the one in the above command is generated when you
+release a project on Github. The above command will generate a file named
+``my_app.rb`` in your current tap. All the project metadata will be filled in
+automatically based on the project on Github. All we need to do now specify
+dependencies and the install commands.
+
+Here's the mostly complete formula for ``my_app``. The tests are left blank as
+these are intended to be integration tests for the app itself that run in an
```

Comment:
It may be worth noting that the tests are usually very basic, just a single command invocation that sanity checks core functionality. Just to not give the wrong impression that users have to write a big integration suite. Maybe link to the homebrew docs on this? https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula